### PR TITLE
Add exception handling in bigint migration

### LINKED ISSUE #6725: Database migration failed - from 6.7.3 to 7.1.0
## Summary

Database migration failed (in init-container, running with `migrate --migrate-to-latest-version` args) upgrading concourse 6.7.3 to 7.1.0.

## Steps to reproduce

Concourse 6.7.3 already deployed and started.
Upgrade concourse to 7.1.0 deploying with concourse helm chart v15.0.1.

## Expected results

Init container (running with `migrate --migrate-to-latest-version` args) successed migrating concourse DB

## Actual results

Init container (running with `migrate --migrate-to-latest-version` args) crashed (almost instantly) with this log: 
` error: migration '1606068653_build_events_partitions_bigint.up.sql' failed and was rolled back: pq: index "pipeline_build_events_14_build_id" does not exist                 │
│ stream closed  `

## Additional context

Helm chart deployment v15.0.1.

## Triaging info

* Concourse version: 7.1.0
* Browser (if applicable): NA
* Did this used to work?

### PR DESCRIPTION
**Pull request by xtremerui**
sumbitting as draft for getting feedbacks of the solution while waiting for user response.

## What does this PR accomplish
Change the migration so it can ignore the error that caused by race condition. We don't really care about the error(table not exist) since ultimately it should be an no-op when pipeline is deleted. 

Bug Fix | Feature | Documentation

fixes #6725 

## Contributor Checklist

- [ ] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [ ] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

File: atc/db/migration/migrations/1606068653_build_events_partitions_bigint.up.sql [hunk 1]
```diff
@@ -7,12 +7,17 @@ BEGIN
...
+    EXECUTE format('ALTER INDEX IF EXISTS pipeline_build_events_%s_build_id_event_id RENAME TO pipeline_build_events_%s_build_id_old_event_id', pipeline.id, pipeline.id);
 
-  RAISE NOTICE 'creating new indexes for pipeline % (%)', pipeline.id, pipeline.name;
-  EXECUTE format('CREATE UNIQUE INDEX pipeline_build_events_%s_build_id_event_id ON pipeline_build_events_%s (build_id, event_id)', pipeline.id, pipeline.id);
+    RAISE NOTICE 'creating new indexes for pipeline % (%)', pipeline.id, pipeline.name;
+    EXECUTE format('CREATE UNIQUE INDEX IF NOT EXISTS pipeline_build_events_%s_build_id_event_id ON pipeline_build_events_%s (build_id, event_id)', pipeline.id, pipeline.id);
+  EXCEPTION
+  WHEN undefined_table then
+    RAISE NOTICE '%', SQLERRM;
+  END;
```
Comment (aoldershaw) [inline atc/db/migration/migrations/1606068653_build_events_partitions_bigint.up.sql:20]: Maybe we can make the exception handling tighter to the commands that could trigger them? i.e. wrap each command in its own try-catch. Maybe the `CREATE UNIQUE INDEX` would need to handle `duplicate_table` instead (just guessing the error code, not sure that's correct)

**Review by aoldershaw — COMMENTED** [review]
I definitely like the addition of `IF NOT EXISTS`, since without it (and without #6727), it makes sense that the migration would continually fail

I wonder if we can make the try-catches tighter to the command that they operate on, rather than running everything in a single try-catch. e.g. if a pipeline is concurrently deleted, if we just handle the exception when we try to `DROP INDEX`, we won't end up modifying the build events indexes and will end up in a bad state

**Comment by xtremerui** [comment]
The `pipeline_build_events_%s_build_id_event_id` index that needs to be modified is also on table `pipeline_build_events_%s` so if that table is gone that index would have been deleted already. 

In such case we could only try-catch the `CREATE UNIQUE INDEX` one with exception `undefined_table`.

**Review by aoldershaw — APPROVED** [review]
Makes sense to me. I think this *technically* won't guarantee the migration will succeed if tables are being deleted concurrently since `IF NOT EXISTS` doesn't enforce synchronization, but there's no precedence for handling that in our migrations

**Pull request merged**